pyproject.toml

```diff
@@ -20,6 +20,7 @@ PyYaml = "^5.1"
 dataclasses = {version = "^0.7", python = "3.6.*"}
 pydantic = "^1.8"
 git-url-parse = "^1"
+typing-extensions = "^4.0.1"
```

Comment:
Since this is a library, you can relax this dependency some? `^4` or `>=4` or something like that. We should probably not specify the build number if we can avoid it.